clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1449,4 +1500,12 @@ final boolean hasUnknownGeneration() {
     final boolean hasValidMemberId() {
         return generation != Generation.NO_GENERATION && generation.hasMemberId();
     }
+
+    final void setNewGeneration(final Generation generation) {
```

Comment:
Should we make these methods synchronized?

Reply:
These functions are only used in non-integration unit tests so I think it is not necessary. LMK if you have a strong motivation?

Reply:
Just because they are exposed. If they ultimately got used in unit tests involving concurrent threads, we would need them to be synchronized. Seems there's no downside to being on the safe side?

Reply:
Yup I think so, let me synchronize this then :)